azkaban-exec-server/src/main/java/azkaban/dag/Node.java

```diff
@@ -39,35 +39,34 @@
 
   private Status status = Status.READY;
 
-  private Dag dag;
+  private final Dag dag;
 
-  Node(final String name, final NodeProcessor nodeProcessor) {
-    this.name = name;
+  Node(final String name, final NodeProcessor nodeProcessor, final Dag dag) {
     requireNonNull(nodeProcessor, "The nodeProcessor parameter can't be null.");
     this.nodeProcessor = nodeProcessor;
+    requireNonNull(name, "The name of the node can't be null");
+    this.name = name;
+    requireNonNull(name, "The dag of the node can't be null");
+    this.dag = dag;
   }
 
-  public Dag getDag() {
+  Dag getDag() {
     return this.dag;
   }
 
-  public void setDag(final Dag dag) {
-    this.dag = dag;
-  }
-
-  private void addParent(final Node node) {
+  /**
+   * Adds the node as the current node's parent i.e. the current node depends on the given node.
+   *
+   * <p>It's important NOT to expose this method as public. The design relies on this to ensure
```

Comment:
Same question regarding the consequences of this scoping.

Reply:
See the answer to the other one.

Reply:
Sounds good!